src/scratchpad/src/org/apache/poi/hsmf/datatypes/MessagePropertiesChunk.java

```diff
@@ -64,28 +87,34 @@ public void readValue(InputStream stream) throws IOException {
         recipientCount = LittleEndian.readUInt(stream);
         attachmentCount = LittleEndian.readUInt(stream);
 
-        // 8 bytes of reserved zeros
-        LittleEndian.readLong(stream);
+        if (!isEmbedded) {
+          // 8 bytes of reserved zeros (top level properties stream only)
+          LittleEndian.readLong(stream);
+        }
 
         // Now properties
         readProperties(stream);
     }
 
     @Override
-    public void writeValue(OutputStream out) throws IOException {
-        // 8 bytes of reserved zeros
-        out.write(new byte[8]);
-
-        // Nexts and counts
-        LittleEndian.putUInt(nextRecipientId, out);
-        LittleEndian.putUInt(nextAttachmentId, out);
-        LittleEndian.putUInt(recipientCount, out);
-        LittleEndian.putUInt(attachmentCount, out);
-
-        // 8 bytes of reserved zeros
+    protected List<PropertyValue> writeHeaderData(OutputStream out) throws IOException
+    {
+      // 8 bytes of reserved zeros
+      out.write(new byte[8]);
+      // Nexts and counts
+      LittleEndian.putUInt(nextRecipientId, out);
+      LittleEndian.putUInt(nextAttachmentId, out);
+      LittleEndian.putUInt(recipientCount, out);
+      LittleEndian.putUInt(attachmentCount, out);
+      // 8 bytes of reserved zeros
+      if (!isEmbedded) {
         out.write(new byte[8]);
+      }
+      // Now properties.
+      return super.writeHeaderData(out);
+    }
 
-        // Now properties
-        writeProperties(out);
+    @Override
+    public void writeValue(OutputStream out) throws IOException {
```

Comment:
I'm sure if I understand why this public method is being emptied of its implementation